libvmi/driver/kvm/kvm.c

```diff
@@ -1115,11 +1115,12 @@ kvm_get_memory_patch(
     }
     else {
         // get the data from kvm
-        nbytes =
-            read(kvm_get_instance(vmi)->socket_fd, buf, length + 1);
-        if (nbytes != (length + 1)) {
+        nbytes = read(kvm_get_instance(vmi)->socket_fd, buf, length + 1);
+        if ( nbytes <= 0 )
```

Comment:
Yea, the problem was comparing signed and unsigned variables but just casting it to unsigned is not safe without first checking if it's less then 0.